eodag/rest/utils/__init__.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023, CS Systemes d'Information, https://www.csgroup.eu/
+#
+# This file is part of EODAG project
+#     https://www.github.com/CS-SI/EODAG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""EODAG REST utils"""
+from __future__ import annotations
+
+import glob
+import logging
+import os
+from io import BufferedReader
+from shutil import make_archive, rmtree
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Iterator,
+    List,
+    NamedTuple,
+    Optional,
+    Tuple,
+)
+from urllib.parse import unquote_plus, urlencode
+
+import dateutil.parser
+import orjson
+from dateutil import tz
+from fastapi import Request
+from pydantic import ValidationError as pydanticValidationError
+
+from eodag.plugins.crunch.filter_latest_intersect import FilterLatestIntersect
+from eodag.plugins.crunch.filter_latest_tpl_name import FilterLatestByName
+from eodag.plugins.crunch.filter_overlap import FilterOverlap
+from eodag.utils import StreamResponse
+from eodag.utils.exceptions import ValidationError
+
+if TYPE_CHECKING:
+    from eodag.rest.types.stac_search import SearchPostRequest
+
+
+logger = logging.getLogger("eodag.rest.utils")
+
+
+class Cruncher(NamedTuple):
+    """Type hinted Cruncher namedTuple"""
+
+    clazz: Callable[..., Any]
+    config_params: List[str]
+
+
+crunchers = {
+    "latestIntersect": Cruncher(FilterLatestIntersect, []),
+    "latestByName": Cruncher(FilterLatestByName, ["name_pattern"]),
+    "overlap": Cruncher(FilterOverlap, ["minimum_overlap"]),
+}
+
+
+def format_pydantic_error(e: pydanticValidationError) -> str:
+    """Format Pydantic ValidationError
+
+    :param e: A Pydantic ValidationError object
+    :tyype e: pydanticValidationError
+    """
+    error_header = f"Invalid request, {e.error_count()} error(s): "
+    error_messages = [err["msg"] for err in e.errors()]
+    return error_header + "; ".join(set(error_messages))
```

Comment:
please send a more detailed description, including the content value that causes the error